### PR TITLE
Precount

### PR DESCRIPTION
**Pull request by idanstark42**

### CODE REVIEW

File: server/configuration.js [hunk 1]
```diff
@@ -15,17 +15,19 @@ const loginPromise = Promise.resolve(mClient.connect())
...
       if (message.topic === CONFIGURATION_TOPIC) {
-        logger.info('Received configurations from mhub')
-        const field = message.data.fields.find(f => f.name === 'clockFormat')
-        if (field) {
-          clockFormat = field.value
-          mhub.sendClockFormat(field.value)
+        logger.info('Received configuration from mhub')
+        const clockFormatField = message.data.clockFormatFields.find(f => f.name === 'clockFormat')
+        const precountField = message.data.clockFormatFields.find(f => f.name === 'precount')
+        if (clockFormatField) {
```
Comment (taltaub22) [inline server/configuration.js]: Why dont you fall back to default format?

File: server/events-linker.js [hunk 2]
```diff
@@ -9,20 +9,22 @@ function catchPromiseWrap (asyncFunction, operationDescription, ...args) {
...
 
+  clockManager.on('prestart', catchPromiseWrap.bind(null, mhubSendPrestartEvent, 'sending prestart event to mhub'))
   clockManager.on('start', catchPromiseWrap.bind(null, mhubSendStartEvent, 'sending start event to mhub'))
   clockManager.on('end', catchPromiseWrap.bind(null, mhubSendEndEvent, 'sending end event to mhub'))
   clockManager.on('stop', catchPromiseWrap.bind(null, mhubSendStopEvent, 'sending stop event to mhub'))
   clockManager.on('reload', catchPromiseWrap.bind(null, mhubSendReloadEvent, 'sending reload event to mhub'))
 
   clockManager.on('start', catchPromiseWrap.bind(null, sounds.playStartSound, 'playing start sound'))
   clockManager.on('stop', catchPromiseWrap.bind(null, sounds.playStopSound, 'playing stop sound'))
+  clockManager.on('end', catchPromiseWrap.bind(null, sounds.playEndSound, 'playing end sound'))
```
Comment (taltaub22) [inline server/events-linker.js]: Could you explain why this was changed?

File: server/configuration.js (hunk 1 above)
Comment (idanstark42) [reply server/configuration.js]: This is a mistake

File: server/events-linker.js (hunk 2 above)
Comment (idanstark42) [reply server/events-linker.js]: This is to make sure the first "0 seconds"event, which occurs when the precount is over, won't play the sound

Comment (idanstark42) [reply server/events-linker.js]: This is to make sure the first "0 seconds"event, which occurs when the precount is over, won't play the sound

File: server/configuration.js (hunk 1 above)
Comment (taltaub22) [reply server/configuration.js]: So this will be fixed?

Comment (idanstark42) [reply server/configuration.js]: The fallback happens in the launcher, on which the module must depend. There was another mistake here, which I fixed.

**Comment by idanstark42** [comment]
We're not, and he would agree with me. We are waiting for a response from @alangreen testing this.

**Pull request merged**